src/palantir/presults.py

```diff
@@ -195,45 +253,79 @@ def _gam_fit_predict_rpy2(x, y, weights=None, pred_x=None):
 
     # Fit the model
     rgam = importr("gam")
-    model = rgam.gam(Formula("y~s(x)"), data=r_df,
-                     weights=pd.Series(weights[use_inds]))
+    model = rgam.gam(Formula("y~s(x)"), data=r_df, weights=pd.Series(weights[use_inds]))
 
     # Predictions
     if pred_x is None:
         pred_x = x
     y_pred = np.array(
         robjects.r.predict(
-            model, newdata=pandas2ri.py2rpy(
-                pd.DataFrame(pred_x, columns=["x"]))
+            model, newdata=pandas2ri.py2rpy(pd.DataFrame(pred_x, columns=["x"]))
         )
     )
 
     # Standard deviations
     p = np.array(
         robjects.r.predict(
-            model, newdata=pandas2ri.py2rpy(
-                pd.DataFrame(x[use_inds], columns=["x"]))
+            model, newdata=pandas2ri.py2rpy(pd.DataFrame(x[use_inds], columns=["x"]))
         )
     )
     n = len(use_inds)
     sigma = np.sqrt(((y[use_inds] - p) ** 2).sum() / (n - 2))
     stds = (
-        np.sqrt(1 + 1 / n + (pred_x - np.mean(x)) **
-                2 / ((x - np.mean(x)) ** 2).sum())
+        np.sqrt(1 + 1 / n + (pred_x - np.mean(x)) ** 2 / ((x - np.mean(x)) ** 2).sum())
         * sigma
         / 2
     )
 
     return y_pred, stds
 
 
-def cluster_gene_trends(trends, k=150, n_jobs=-1):
-    """Function to cluster gene trends
-    :param trends: Matrix of gene expression trends
-    :param k: K for nearest neighbor construction
-    :param n_jobs: Number of jobs for parallel processing
-    :return: Clustering of gene trends
+def cluster_gene_trends(
```

Comment:
Can we please add an optional `varm` key if Anndata is specified to indicate which genes need to be clustered? I dont think we would want to cluster all genes at any point.

Reply:
With bb579cc72e3c6594530624ea3ca5bed9c07725bb the function now takes a gene list in the `genes` argument.